### DISCUSSION TITLE
Reverting back to tuya... not enough space

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#18675] [General]
I converted a device from tuya to tasmota only to find out that the relays don't work. 
I wanted to convert it back so I could send it back but using the backup shows 'not enough space'. 

-rw-r--r--  1 root root 1048576 May 16 16:12 firmware-cc3841.bin

The thing is, I re-flashed it using the minimal firmware before doing that but it still doesn't work.
I've read that it's not possible to go back to tuya without using serial which in my case, would not allow me to send it back.

Is there something I am missing or it just cannot be done? If I have to junk it, so be it but wanted to ask first.

**Comment by sfromis** [comment]
Reverting to a firmware backup file is indeed only possible via serial. It may be possible to get the device to work with Tasmota, if doing the proper config.

**Comment by dac2020** [comment]
>It may be possible to get the device to work with Tasmota, if doing the proper config.

I assume you mean using the gui to re-install the backup?

That's what I was wondering also, the proper config meaning trimming it down as much as possible. I've been looking at the commands page to see what I might be able to slim down. I assume this has to be done after installing the minimal firmware.

Not much info on the net on how to trim down even more however.

**Reply by sfromis** [reply]
If you have the tasmota-minimal firmware installed, nothing works. What's need for operation is `tasmota.bin`.

No, "proper config" is not at all about "trimming down", but abut doing the right config for the hardware. Most Tuya devices with ESP8266 have an additional chip to control the device (the TuyaMCU we call it), with serial communication to the ESP8266, meaning that it needs to have the correct tx/rx pins configured (if not the default pins), with the right baud rate. In some cases, additional Tasmota commands are needed to enable all available functions.

**Reply by dac2020** [reply]
Ok, then I'll junk it. If I could have converted it back, I would not feel bad about sending it back but since it cannot be without taking it apart, then I'll just junk it since the relays don't work on it anyhow.

Thanks.
